src/main/java/com/google/devtools/build/lib/rules/cpp/LibrariesToLinkCollector.java

```diff
@@ -607,9 +607,9 @@ private void addDynamicInputLinkOptions(
     // -l:foo -> foo.so
     // -l:libfoo.so.1 -> libfoo.so.1
     boolean hasCompatibleName =
-        name.startsWith("lib") || (!name.endsWith(".so") && !name.endsWith(".dylib"));
+        name.startsWith("lib") || (!name.endsWith(".so") && !name.endsWith(".dylib") && !name.endsWith(".dll"));
```

Comment:
Looks strange for me... Logical operators are misplaced here. Thats why if You pass `lib\<something\>.so` it works fine, but if You pass `\<something\>.so` linker complains not finding file `\<something\>.so` because `.so` is not stripped. Should it be:
`name.startsWith("lib") || (name.endsWith(".so") || name.endsWith(".dylib")  || name.endsWith(".dll"));` ?